library/src/test/java/com/github/pwittchen/reactivenetwork/library/internet/observing/strategy/WalledGardenInternetObservingStrategyTest.java

```diff
@@ -81,58 +116,80 @@ private String getHost() {
+  @Test public void shouldCreateHttpsUrlConnection() throws IOException {
     // given
-    final String givenHost = "http://www.website.com";
+    final String parsedDefaultHost = "clients3.google.com";
 
     // when
-    String transformedHost = strategy.adjustHost(givenHost);
+    HttpURLConnection connection =
+        strategy.createHttpsUrlConnection(getHost(), PORT, TIMEOUT_IN_MS);
 
     // then
-    assertThat(transformedHost).isEqualTo(givenHost);
+    assertThat(connection).isNotNull();
+    assertThat(connection.getURL().getHost()).isEqualTo(parsedDefaultHost);
+    assertThat(connection.getURL().getPort()).isEqualTo(PORT);
+    assertThat(connection.getConnectTimeout()).isEqualTo(TIMEOUT_IN_MS);
+    assertThat(connection.getReadTimeout()).isEqualTo(TIMEOUT_IN_MS);
+    assertThat(connection.getInstanceFollowRedirects()).isFalse();
+    assertThat(connection.getUseCaches()).isFalse();
   }
 
-  @Test public void shouldNotTransformHttpsHost() {
+  @Test public void shouldHandleAnExceptionWhileCreatingHttpsUrlConnection() throws IOException {
     // given
-    final String givenHost = "https://www.website.com";
+    final String errorMsg = "Could not establish connection with WalledGardenStrategy";
+    final IOException givenException = new IOException(errorMsg);
+    when(strategy.createHttpsUrlConnection(getHost(), PORT, TIMEOUT_IN_MS)).thenThrow(
+        givenException);
 
     // when
-    String transformedHost = strategy.adjustHost(givenHost);
+    strategy.isConnected(getHost(), PORT, TIMEOUT_IN_MS, errorHandler);
 
     // then
-    assertThat(transformedHost).isEqualTo(givenHost);
+    verify(errorHandler).handleError(givenException, errorMsg);
   }
 
-  @Test public void shouldAddHttpProtocolToHost() {
-    // given
-    final String givenHost = "www.website.com";
-    final String expectedHost = "http://www.website.com";
+  @Test public void shouldNotTransformHttpHost() {
+    // when
+    String transformedHost = strategy.adjustHost(HOST_WITH_HTTPS);
+
+    // then
+    assertThat(transformedHost).isEqualTo(HOST_WITH_HTTPS);
+  }
+
+  @Test public void shouldNotTransformHttpsHost() {
+    // when
+    String transformedHost = strategy.adjustHost(HOST_WITH_HTTPS);
+
+    // then
+    assertThat(transformedHost).isEqualTo(HOST_WITH_HTTPS);
+  }
 
+  @Test public void shouldAddHttpsProtocolToHost() {
     // when
-    String transformedHost = strategy.adjustHost(givenHost);
+    String transformedHost = strategy.adjustHost(HOST_WITHOUT_HTTPS);
 
     // then
-    assertThat(transformedHost).isEqualTo(expectedHost);
+    assertThat(transformedHost).isEqualTo(HOST_WITH_HTTPS);
   }
 
   @Test public void shouldAdjustHostWhileCheckingConnectivity() {
     // given
     final String host = getHost();
-    when(strategy.isConnected(host, PORT, TIMEOUT_IN_MS, errorHandler)).thenReturn(true);
+    Mockito.doReturn(true).when(strategy).isConnected(host, PORT, TIMEOUT_IN_MS, errorHandler);
```

Comment:
Please, use static import to exclude `Mockito` class name. It should look like: `doReturn(true).when(...)`. All tests in this project use that convention.